pkg/policies/flowcontrol/service/envoy/authz.go

```diff
@@ -253,6 +253,14 @@ func (h *Handler) Check(ctx context.Context, req *authv3.CheckRequest) (*authv3.
 					},
 				},
 			}
+		case flowcontrolv1.CheckResponse_REJECT_REASON_FLOW_REGULATED:
+			resp.HttpResponse = &authv3.CheckResponse_DeniedResponse{
+				DeniedResponse: &authv3.DeniedHttpResponse{
+					Status: &typev3.HttpStatus{
+						Code: typev3.StatusCode_Unauthorized,
+					},
+				},
+			}
 		default:
 			log.Bug().Stringer("reason", checkResponse.RejectReason).Msg("Unexpected reject reason")
 		}
```

Comment:
Too many requests as the status code might also mislead.
The FlowRegulator component drops a certain percentage of requests either randomly or based on consistent hashing of session ids.
Based on my description, do you recommend changing FlowRegulator to something else? Also, what would you recommend as the status code for requests that get dropped because of this criteria?